### PR TITLE
KOGITO-4266: [DMN Designer] Decision Service is missing inputData element in model with multiple DRDs

### PR DESCRIPTION
**Pull request by danielzhe**
**JIRA**: [KOGITO-4266](https://issues.redhat.com/browse/KOGITO-4266)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by danielzhe** [comment]
jenkins do fdb

**Comment by danielzhe** [comment]
jenkins do fdb

**Comment by danielzhe** [comment]
Working on test failures.

**Comment by danielzhe** [comment]
jenkins do fdb

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/converters/DecisionServiceConverter.java [hunk 1]
```diff
@@ -267,9 +280,13 @@ private void inspectDecisionForDSReqs(final Node<View<?>, ?> targetNode,
...
                     if (drgElement instanceof Decision) {
-                        reqDecisions.add((Decision) drgElement);
+                        if (!reqDecisions.contains(drgElement)) {
+                            reqDecisions.add((Decision) drgElement);
+                        }
                     } else if (drgElement instanceof InputData) {
-                        reqInputs.add((InputData) drgElement);
+                        if (!reqInputs.contains(drgElement)) {
+                            reqInputs.add((InputData) drgElement);
+                        }
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/converters/DecisionServiceConverter.java:289]: Is we call quite often `contains` method for both `reqDecisions` and `reqInputs`, I think we could change them from `ArraysList` to `HashSet` for enhancing a performance.

**Review by jomarko — COMMENTED** [review]
Added @jstastny-cz as reviewer, as he is a tester in jira and also the reporter, so knows more details about the issue as me. Just one comment to code from me. Thank you for a PR @danielzhe .

The final review will be provided by @jstastny-cz .

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/converters/DecisionServiceConverter.java (hunk 1 above)
Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/converters/DecisionServiceConverter.java:289]: Thank you, that's a good point.

**Comment by karreiro** [comment]
@danielzhe is this PR ready for the re-review? :-)

**Comment by danielzhe** [comment]
yes, @karreiro

**Comment by jstastny-cz** [comment]
jenkins do FDB

**Comment by jstastny-cz** [comment]
@danielzhe I miss insight into the solution here. In the Jira there's the assumption that inputData element might be missing from the resulting DecisionService definition. When I modelled the decision service using now, the inputData element is still not in the xml, but evaluation seems to work correctly. Please comment on the Jira with the solution at hand - especially on the incorrect assumption.

**Comment by karreiro** [comment]
@jstastny-cz IIUC, the buggy scenario that [KOGITO-4266](https://issues.redhat.com/browse/KOGITO-4266) solves is this one:

- users create a **Decision Service A**
- users add a **Decision B** inside of the **Decision Service A**
- users create a new **DRD C**
- users add the **Decision B** inside of the **DRD C**
- users connect an **Input Data D** into the **Decision B** (in the **DRD C**)

Before this PR, the **Input Data D** was not appearing in the properties panel when the **Decision Service A** was selected. Also, and the **Input Data D** reference was not appearing as an `inputData` in the Decision Service XML.

I tried it locally, and it seems to work on both scenarios. Am I missing something?

**Review by jstastny-cz — APPROVED** [review]
Solved in chat, thank you for your insights.

The feature works as expected indeed and description in Jira is in sync with the solution provided here.

**Comment by karreiro** [comment]
Thank you, @jstastny-cz! :-)

**Pull request merged**